rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -959,6 +936,43 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 		propagate_used_devices();
 	};
 
+	const auto get_audio_output_devices = [this]()
+	{
+		auto dev_enum = Emu.GetCallbacks().get_audio_enumerator();
+		std::vector<audio_device_enumerator::audio_device> dev_array = dev_enum->get_output_devices();
+
+		ui->audioDeviceBox->clear();
+		ui->audioDeviceBox->blockSignals(true);
+		ui->audioDeviceBox->addItem(tr("Default"), qsv(audio_device_enumerator::DEFAULT_DEV_ID));
+
+		int device_index = 0;
+
+		for (auto& dev : dev_array)
+		{
+			const QString cur_item = qstr(dev.name);
+			ui->audioDeviceBox->addItem(cur_item, qstr(dev.id));
+			if (g_cfg.audio.audio_device.to_string() == dev.id)
+			{
+				device_index = ui->audioDeviceBox->findText(cur_item);
```

Comment:
I think this is safer:
```suggestion
				device_index = ui->audioDeviceBox->findData(qstr(dev.id));
```